junit5/src/main/java/org/quickperf/junit5/QuickPerfTestExtension.java

```diff
@@ -52,20 +59,59 @@ public void interceptTestMethod(Invocation<Void> invocation,
         List<RecordablePerformance> perfRecordersToExecuteBeforeTestMethod = testExecutionContext.getPerfRecordersToExecuteBeforeTestMethod();
         List<RecordablePerformance> perfRecordersToExecuteAfterTestMethod = testExecutionContext.getPerfRecordersToExecuteAfterTestMethod();
 
-        if(!testExecutionContext.isQuickPerfDisabled()) {
-            startRecordings(perfRecordersToExecuteBeforeTestMethod);
+        if(SystemProperties.TEST_CODE_EXECUTING_IN_NEW_JVM.evaluate()) {
+            if (!testExecutionContext.isQuickPerfDisabled()) {
+                startRecording(perfRecordersToExecuteBeforeTestMethod);
+            }
+            try {
+                invocation.proceed();
+                //FIXME call directly the test, this needs to allow to skip the invocation (see bellow)
+                //invocationContext.getExecutable().invoke(invocationContext.getArguments());
+                //invocation.skip();
+            } finally {
+                if (!testExecutionContext.isQuickPerfDisabled()) {
+                    stopRecording(perfRecordersToExecuteAfterTestMethod);
+                }
+            }
+            return;
         }
 
         Throwable businessThrowable = null;
-        try{
+        if (testExecutionContext.testExecutionUsesTwoJVMs()) {
+            newJvmTestLauncher.run( invocationContext.getExecutable()
+                    , testExecutionContext.getWorkingFolder()
+                    , testExecutionContext.getJvmOptions()
+                    , QuickPerfJunit5Core.class);
+
+            //FIXME we need to invoke it here and in the fork because Junit check that the invocation was proceed
+            // this causes double invocation of the test !!!
+            // this should be fixed in JUnit 5.6 that will provides a way to skip an invocation: https://github.com/junit-team/junit5/pull/2090
+            //invocation.skip();
             invocation.proceed();
+
+            WorkingFolder workingFolder = testExecutionContext.getWorkingFolder();
+            businessThrowable = jUnit5FailuresRepository.find(workingFolder);
         }
-        catch (Throwable throwable){
-            businessThrowable = throwable;
-        } finally {
-            if (!testExecutionContext.isQuickPerfDisabled()) {
-                stopRecordings(perfRecordersToExecuteAfterTestMethod);
+        else {
+            try{
+                if(!testExecutionContext.isQuickPerfDisabled()){
+                    startRecording(perfRecordersToExecuteBeforeTestMethod);
+                }
+                invocation.proceed();
+            }
+            catch (Throwable throwable){
+                businessThrowable = throwable;
             }
+            finally {
+                if(!testExecutionContext.isQuickPerfDisabled()){
+                    stopRecording(perfRecordersToExecuteAfterTestMethod);
+                }
+            }
+        }
+
+        //FIXME why didn't the ConsoleReporter nor JUnit shows the throwable ?
```

Comment:
It is OK: issueThrower.throwIfNecessary(businessThrowable, groupOfPerfIssuesToFormat);
I created a PR to check this with an automatic test

Reply:
This was a bug, I forget to print the failure report.
This is fixed now.